build-tools/packages/build-cli/src/library/changesets.ts

```diff
@@ -3,42 +3,112 @@
  * Licensed under the MIT License.
  */
 
+import { readFile } from "node:fs/promises";
 import path from "node:path";
 import { VersionBumpType } from "@fluid-tools/version-tools";
-import { Logger } from "@fluidframework/build-tools";
+import { Logger, type ReleaseNotesSection } from "@fluidframework/build-tools";
 import { compareAsc, formatISO, parseISO } from "date-fns";
 import globby from "globby";
 import matter from "gray-matter";
+import issueParser from "issue-parser";
+const { test: hasFrontMatter } = matter;
 
 import { ReleasePackage } from "../releaseGroups.js";
 import { Repository } from "./git.js";
 
 export const DEFAULT_CHANGESET_PATH = ".changeset";
 
+/**
+ * The section name used for changesets that do not match any defined sections.
+ */
+export const UNKNOWN_SECTION = "_unknown";
+
+/**
+ * Additional metadata that can be used inside a changeset. This metadata should be set in a second fron-matter section.
+ * For example:
+ *
+ * ```
+ * ---
+ * "package-a": minor
+ * ---
+ * ---
+ * section: fix
+ * ---
+ *
+ * Changeset title.
+ *
+ * Changeset details
+ * ```
+ */
+export interface FluidCustomChangesetMetadata {
+	/**
+	 * The section in release notes in which this changeset should be included.
+	 */
+	section?: ReleaseNotesSection["name"];
+
+	/**
+	 * If false, the changeset will not be included in release notes. Defaults to true.
+	 */
+	includeInReleaseNotes?: boolean;
+
+	/**
+	 * If true, the changeset will be ordered before the other changes in that section. If multiple changesets are
```

Comment:
Is the default value `false`? We should document what `undefined` means. Same for the other optional properties.